File: EPIC.Api/Models/ChurchService.cs

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EPIC.Api.Models
{
    public class ChurchService
    {
        // =========================================================
        // PRIMARY KEY
        // =========================================================

        [Key]
        public int ChurchServiceId { get; set; }

        // =========================================================
        // CUSTOMER / TENANT
        // =========================================================

        [Required]
        public int CustomerId { get; set; }

        [ForeignKey(nameof(CustomerId))]
        public Customer? Customer { get; set; }

        // =========================================================
        // SERVICE INFORMATION
        // =========================================================

        [Required]
        [MaxLength(150)]
        public string ServiceName { get; set; } = string.Empty;

        [MaxLength(50)]
        public string ServiceType { get; set; } = string.Empty;

        [Required]
        public DateTime ServiceDate { get; set; }

        [MaxLength(50)]
        public string StartTime { get; set; } = string.Empty;

        [MaxLength(50)]
        public string EndTime { get; set; } = string.Empty;

        // =========================================================
        // LOCATION
        // =========================================================

        [MaxLength(200)]
        public string Location { get; set; } = string.Empty;

        // =========================================================
        // LEADER / SPEAKER
        // =========================================================

        [MaxLength(150)]
        public string ServiceLeader { get; set; } = string.Empty;

        [MaxLength(150)]
        public string Speaker { get; set; } = string.Empty;

        // =========================================================
        // DESCRIPTION
        // =========================================================

        [MaxLength(500)]
        public string Description { get; set; } = string.Empty;

        // =========================================================
        // STATUS
        // =========================================================

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "SCHEDULED";

        // =========================================================
        // AUDIT
        // =========================================================

        [Required]
        public DateTime CreatedDate { get; set; } = DateTime.Now;

        public DateTime? UpdatedDate { get; set; }
    }
}

File: EPIC.Web/src/services/churchService.ts
import { API_BASE_URL } from "../config";

export interface ChurchService {
    churchServiceId: number;
    serviceName: string;
    serviceType: string;
    serviceDate: string;
    startTime: string;
    endTime: string;
    location: string;
    serviceLeader: string;
    speaker: string;
    description: string;
    status: string;
}

export async function getUpcomingChurchServices(
    limit: number = 50,
    includePast: boolean = false
): Promise<ChurchService[]> {
    const queryParams = new URLSearchParams({
        limit: limit.toString(),
        includePast: includePast.toString()
    });

    const targetUrl = `${API_BASE_URL}/ChurchServices/public/upcoming?${queryParams.toString()}`;

    try {
        const response = await fetch(targetUrl, {
            method: "GET",
            headers: {
                Accept: "application/json"
            }
        });

        if (!response.ok) {
            const error = await response.text();
            console.error("Church Service API Error:", response.status, error);
            throw new Error(`Unable to load church services (${response.status})`);
        }

        return await response.json();
    } catch (err) {
        console.warn("Primary Church Service endpoint failed, attempting fallback:", err);
        // Fallback for local Vite dev if API_BASE_URL was different
        const fallbackUrl = `http://localhost:5109/api/ChurchServices/public/upcoming?${queryParams.toString()}`;
        if (targetUrl !== fallbackUrl) {
            const fallbackRes = await fetch(fallbackUrl, {
                method: "GET",
                headers: { Accept: "application/json" }
            });
            if (fallbackRes.ok) {
                return await fallbackRes.json();
            }
        }
        throw err;
    }
}